### PR TITLE
Update support for loadable_apps on RTL8730E

### PR DESCRIPTION
**Pull request by amogh-samsung**
- Fix issues with common binary working
- Fix build issues with build for rtl8730e/loadable_apps

### CODE REVIEW

File: os/arch/boardctl.c [hunk 1]
```diff
@@ -71,6 +71,14 @@
  * Public Functions
  ****************************************************************************/
 
+#ifdef CONFIG_ARCH_BOARD_RTL8730E
+// TODO: Added this dummy API to fix build error. Remove after rtk provides implementation
+int board_reset(int x)
```
Comment (kishore-sn) [inline os/arch/boardctl.c:76]: Since we are adding this in boot.c let us remove this function from this file.

Comment (amogh-samsung) [reply os/arch/boardctl.c]: Updated,
Thankyou

File: os/arch/arm/src/armv7-a/arm_reprioritizertr.c [hunk 2]
```diff
@@ -153,6 +153,8 @@ void up_reprioritize_rtr(struct tcb_s *tcb, uint8_t priority)
 
               rtcb = this_task();
 
+	  up_restoretask(rtcb);
```
Comment (jeongarmy) [inline os/arch/arm/src/armv7-a/arm_reprioritizertr.c:156]: Please check the alignment.

File: os/arch/boardctl.c [hunk 3]
```diff
@@ -150,7 +150,10 @@ int boardctl(unsigned int cmd, uintptr_t arg)
 		sched_lock();
 		/* Add 100ms delay for flushing another logs like printf. */
 		up_mdelay(100);
+#ifndef CONFIG_ARCH_BOARD_RTL8730E
```
Comment (sunghan-chang) [inline os/arch/boardctl.c:153]: Could you let me know detail?

File: os/board/rtl8730e/src/rtl8730e_boot.c [hunk 4]
```diff
@@ -273,8 +273,8 @@ void amebasmart_mount_partitions(void)
 }
 
 #ifdef CONFIG_FTL_ENABLED
-extern const u8 ftl_phy_page_num;
-extern const u32 ftl_phy_page_start_addr;
+extern u8 ftl_phy_page_num;
+extern u32 ftl_phy_page_start_addr;
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/rtl8730e_boot.c:277]: Could you let me know why this is necessary?

File: os/arch/arm/src/armv7-a/arm_blocktask.c [hunk 5]
```diff
@@ -130,6 +130,7 @@ void up_block_task(struct tcb_s *tcb, tstate_t task_state)
            */
 
           rtcb = this_task();
+	  up_restoretask(rtcb);
```
Comment (sunghan-chang) [inline os/arch/arm/src/armv7-a/arm_blocktask.c:133]: Please change all indentation with tab at different PR.

File: os/arch/arm/src/armv7-a/arm_syscall.c [hunk 6]
```diff
@@ -287,8 +287,10 @@ uint32_t *arm_syscall(uint32_t *regs)
 			* set will determine the restored context.
 			*/
 
+		struct tcb_s *rtcb = sched_self();
 		CURRENT_REGS = (uint32_t *)regs[REG_R1];
 		DEBUGASSERT(CURRENT_REGS);
+	  up_restoretask(rtcb);
```
Comment (sunghan-chang) [inline os/arch/arm/src/armv7-a/arm_syscall.c:293]: alignment

File: os/arch/arm/src/armv7-a/arm_syscall.c [hunk 7]
```diff
@@ -311,9 +313,11 @@ uint32_t *arm_syscall(uint32_t *regs)
 
 	case SYS_switch_context:
 	 {
+		struct tcb_s *rtcb = sched_self();
 		DEBUGASSERT(regs[REG_R1] != 0 && regs[REG_R2] != 0);
 		*(uint32_t **)regs[REG_R1] = regs;
 		CURRENT_REGS = (uint32_t *)regs[REG_R2];
+	  up_restoretask(rtcb);
```
Comment (sunghan-chang) [inline os/arch/arm/src/armv7-a/arm_syscall.c:320]: same as previous

File: os/arch/arm/src/armv7-a/arm_blocktask.c (hunk 5 above)
Comment (jeongarmy) [inline os/arch/arm/src/armv7-a/arm_blocktask.c:133]: Please the comments for calling up_restoretask. `/* Restore rtcb data for context switching */`

File: os/arch/boardctl.c (hunk 3 above)
Comment (amogh-samsung) [reply os/arch/boardctl.c]: This change is not required post https://github.com/Samsung/TizenRT/pull/5868

File: os/board/rtl8730e/src/rtl8730e_boot.c (hunk 4 above)
Comment (amogh-samsung) [reply os/board/rtl8730e/src/rtl8730e_boot.c:277]: The ftl physical page number and start addresses are updated by some line of code (Eg `board/rtl8730e/src/component/usrcfg/rtl8730e/ameba_flashcfg.c`) and should not be declared as constant variable types.

File: os/arch/arm/src/armv7-a/arm_syscall.c (hunk 7 above)
Comment (amogh-samsung) [reply os/arch/arm/src/armv7-a/arm_syscall.c]: Done

File: os/arch/arm/src/armv7-a/arm_reprioritizertr.c (hunk 2 above)
Comment (amogh-samsung) [reply os/arch/arm/src/armv7-a/arm_reprioritizertr.c]: Done

File: os/arch/arm/src/armv7-a/arm_blocktask.c (hunk 5 above)
Comment (amogh-samsung) [reply os/arch/arm/src/armv7-a/arm_blocktask.c]: Done already

**Review by sunghan-chang — APPROVED** [review]

File: os/arch/arm/src/armv7-a/arm_unblocktask.c [hunk 8]
```diff
@@ -113,6 +113,9 @@ sched_removeblocked(tcb);
            */
 
           rtcb = this_task();
+
+	  /* Restore rtcb data for context switching */
```
Comment (sunghan-chang) [inline os/arch/arm/src/armv7-a/arm_unblocktask.c:117]: This kind of change should be in 1st commit, not 2nd commit.

Comment (amogh-samsung) [reply os/arch/arm/src/armv7-a/arm_unblocktask.c:117]: Updated, thankyou

**Pull request merged**